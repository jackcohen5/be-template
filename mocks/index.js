// COMMON
export const mockUserId1 = 'user_id_1'

// LAMBDA
export const authenticatedEvent = {
    requestContext: { authorizer: { userId: mockUserId1 } },
}

// ITEM
export const mockItem1 = {
    itemId: 'item_id_1',
    userId: mockUserId1,
}

export const mockItem2 = {
    itemId: 'item_id_2',
    userId: 'user_id_2',
}

export const mockItems = [mockItem1, mockItem2]
