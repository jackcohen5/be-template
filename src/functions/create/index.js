import uuid from 'uuid'

import BaseView from 'functions/BaseView'
import { put as PutItem } from 'services/DynamoDB'

const CreateView = async ({ userId }) => {
    const itemId = uuid.v4()
    const item = await PutItem(userId, itemId)
    return { data: item, statusCode: 201 }
}

export default BaseView(CreateView)
