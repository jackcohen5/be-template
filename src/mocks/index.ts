import {
    APIGatewayEventRequestContext,
    APIGatewayProxyEvent,
    APIGatewayProxyEventBase,
    APIGatewayProxyEventPathParameters,
} from 'aws-lambda'

import { Roles } from 'functions/authorizer/authorizer.constants'

// COMMON
export const mockUserId1 = 'user_id_1'

// LAMBDA
const baseEvent: APIGatewayProxyEvent = {
    body: '',
    headers: undefined,
    multiValueHeaders: undefined,
    httpMethod: '',
    isBase64Encoded: false,
    path: '',
    pathParameters: undefined,
    queryStringParameters: undefined,
    multiValueQueryStringParameters: undefined,
    stageVariables: undefined,
    requestContext: undefined,
    resource: '',
}

const baseResourceContext: APIGatewayEventRequestContext = {
    accountId: '',
    apiId: '',
    authorizer: undefined,
    protocol: '',
    httpMethod: '',
    identity: undefined,
    path: '',
    stage: '',
    requestId: '',
    requestTimeEpoch: 0,
    resourceId: '',
    resourcePath: '',
}

export const makeEvent = <TAuthorizerContext>({
    authorizer,
    body = '{}',
    pathParameters = {},
}: {
    body?: string
    pathParameters?: APIGatewayProxyEventPathParameters
    authorizer?: TAuthorizerContext
} = {}): APIGatewayProxyEventBase<TAuthorizerContext> => ({
    ...baseEvent,
    requestContext: {
        ...baseResourceContext,
        authorizer,
    },
    body: JSON.stringify(body),
    pathParameters,
})

export const makeAuthenticatedEvent = ({
    body,
    pathParameters,
    roles = [Roles.TEMPLATE_NAME_ROLE1],
}: {
    body?: string
    pathParameters?: APIGatewayProxyEventPathParameters
    roles?: Roles[]
} = {}): APIGatewayProxyEvent =>
    makeEvent({
        authorizer: { userId: mockUserId1, roles },
        body,
        pathParameters,
    })

export const role1AuthenticatedEvent = makeAuthenticatedEvent()

// ITEM
export const mockItem1 = {
    itemId: 'item_id_1',
    userId: mockUserId1,
}

export const mockItem2 = {
    itemId: 'item_id_2',
    userId: 'user_id_2',
}

export const mockItems = [mockItem1, mockItem2]
