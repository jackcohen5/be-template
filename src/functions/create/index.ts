import { v4 as uuidv4 } from 'uuid'

import { FunctionHandler, Role1View } from 'functions/BaseView'
import { BaseItem, put as PutItem } from 'services/DynamoDB'

interface CreatePayload {
    item: BaseItem
}
const CreateView: FunctionHandler<CreatePayload> = async ({
    auth: { userId },
    body: data,
}) => {
    const TEMPLATE_NAME_SK = uuidv4()
    const pk = `USER#${userId}`
    const sk = `ORDER#${TEMPLATE_NAME_SK}`
    const item = await PutItem({ pk, sk, data })
    return { data: { item }, statusCode: 201 }
}

export default Role1View(CreateView)
