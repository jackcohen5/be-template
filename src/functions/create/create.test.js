import { authenticatedEvent } from '../../../mocks'

import create from '.'

describe('create', () => {
    it('executes as expected', async () => {
        const response = await create(authenticatedEvent)
        expect(response).toMatchSnapshot()
    })
})
