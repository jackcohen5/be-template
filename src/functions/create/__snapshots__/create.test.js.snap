// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create executes as expected 1`] = `
Object {
  "body": "{\\"data\\":{\\"itemId\\":\\"item_id_1\\",\\"userId\\":\\"user_id_1\\"}}",
  "headers": Object {
    "Access-Control-Allow-Origin": "*",
  },
  "statusCode": 201,
}
`;
